src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -409,102 +409,91 @@ private ValueGetter<uint> ComposeGetterOne(Row input, int iinfo, int srcCol, Col
 
             if (srcType is KeyType)
             {
-                switch (srcType.RawKind)
-                {
-                    case DataKind.U1:
-                        return MakeScalarHashGetter<byte, HashKey1>(input, srcCol, seed, mask);
-                    case DataKind.U2:
-                        return MakeScalarHashGetter<ushort, HashKey2>(input, srcCol, seed, mask);
-                    case DataKind.U4:
-                        return MakeScalarHashGetter<uint, HashKey4>(input, srcCol, seed, mask);
-                    default:
-                        Host.Assert(srcType.RawKind == DataKind.U8);
-                        return MakeScalarHashGetter<ulong, HashKey8>(input, srcCol, seed, mask);
-                }
+                if (srcType.RawType == typeof(byte))
+                    return MakeScalarHashGetter<byte, HashKey1>(input, srcCol, seed, mask);
+                else if (srcType.RawType == typeof(ushort))
+                    return MakeScalarHashGetter<ushort, HashKey2>(input, srcCol, seed, mask);
+                else if (srcType.RawType == typeof(uint))
+                    return MakeScalarHashGetter<uint, HashKey4>(input, srcCol, seed, mask);
+
```

Comment:
By a similar token, I might want to check for `uint` first, since that is going to be the key type 99% of the time. :)

Reply:
Fixed.

---
In reply to: [248937303](https://github.com/dotnet/machinelearning/pull/2176#discussion_r248937303) [](ancestors = 248937303)